<a href="https://colab.research.google.com/github/najla-alhusaini/it326-telco-churn/blob/main/Reports/Phase1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

from pathlib import Path

# Read the local copy when the repo is cloned; fall back to GitHub (e.g. in Colab)
local = next((p for p in [Path("../Dataset/Raw_dataset.csv"), Path("Dataset/Raw_dataset.csv")] if p.exists()), None)
url = local or 'https://raw.githubusercontent.com/najla-alhusaini/it326-telco-churn/main/Dataset/Raw_dataset.csv'
df = pd.read_csv(url)

print("Shape (rows, cols):", df.shape)
print("\nData types:\n", df.dtypes)

TARGET_COLUMN = 'Churn'
if TARGET_COLUMN in df.columns:
    print("\nClass counts:")
    print(df[TARGET_COLUMN].value_counts())
else:
    print(f"Target column '{TARGET_COLUMN}' not found!")

print("\nSample of dataset:")
display(df.head())

Shape (rows, cols): (7043, 21)

Data types:
 customerID              str
gender                  str
SeniorCitizen         int64
Partner                 str
Dependents              str
tenure                int64
PhoneService            str
MultipleLines           str
InternetService         str
OnlineSecurity          str
OnlineBackup            str
DeviceProtection        str
TechSupport             str
StreamingTV             str
StreamingMovies         str
Contract                str
PaperlessBilling        str
PaymentMethod           str
MonthlyCharges      float64
TotalCharges            str
Churn                   str
dtype: object

Class counts:
Churn
No     5174
Yes    1869
Name: count, dtype: int64

Sample of dataset:


,customerID,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,OnlineBackup,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,MonthlyCharges,TotalCharges,Churn
0,7590-VHVEG,Female,0,Yes,No,1,No,No phone service,DSL,No,Yes,No,No,No,No,Month-to-month,Yes,Electronic check,29.85,29.85,No
1,5575-GNVDE,Male,0,No,No,34,Yes,No,DSL,Yes,No,Yes,No,No,No,One year,No,Mailed check,56.95,1889.5,No
2,3668-QPYBK,Male,0,No,No,2,Yes,No,DSL,Yes,Yes,No,No,No,No,Month-to-month,Yes,Mailed check,53.85,108.15,Yes
3,7795-CFOCW,Male,0,No,No,45,No,No phone service,DSL,Yes,No,Yes,Yes,No,No,One year,No,Bank transfer (automatic),42.30,1840.75,No
4,9237-HQITU,Female,0,No,No,2,Yes,No,Fiber optic,No,No,No,No,No,No,Month-to-month,Yes,Electronic check,70.70,151.65,Yes


# Phase 1 - Dataset Overview

## Project Goal
The goal of this project is to **classify telecom customers** based on whether they will churn (leave the company) or not.  
- **Classification Task:** Predict the target attribute `Churn` (Yes / No) using customer demographic, service, and billing information.  
- **Clustering Task:** Group customers into segments based on their attributes to discover patterns in customer behavior and identify groups with higher churn risk.
- These tasks help telecom companies identify at-risk customers and design targeted retention strategies for different customer segments
## Dataset Source
- **Source:** Kaggle – [Telco Customer Churn Dataset](https://www.kaggle.com/datasets/blastchar/telco-customer-churn)  
- **File:** `Dataset/Raw_dataset.csv`

## Dataset Description
- The dataset contains 21 attributes with mixed data types: categorical (e.g., gender, Contract, PaymentMethod), numerical (SeniorCitizen, tenure, MonthlyCharges), and one object-type column (TotalCharges) that requires conversion to numeric due to non-standard entries.

- It includes 7,043 records (instances) representing customer information.

- The class attribute is Churn, which indicates whether a customer left the company. The distribution is:

	•	No (Non-churn): 5,174 instances
	•	Yes (Churn): 1,869 instances


